docs/docs/render-props.md

```diff
@@ -0,0 +1,224 @@
+    this.setState({
+      x: event.clientX,
+      y: event.clientY
+    });
+  }
+
+  render() {
+    return (
+      <div style={{ height: '100%' }} onMouseMove={this.handleMouseMove}>
+
+        {/* ...but how do we render something other than a <p>? */}
+        <p>The current mouse position is ({this.state.x}, {this.state.y})</p>
+      </div>
+    );
+  }
+}
+
+class MouseTracker extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Move the mouse around!</h1>
+        <Mouse/>
+      </div>
+    );
+  }
+}
+```
+
+Now the `<Mouse>` component encapsulates all behavior associated with listening for `mousemove` events and storing the (x, y) position of the cursor, but it's not yet truly reusable.
+
+For example, let's say we have a `<Cat>` component that renders the image of a cat chasing the mouse around the screen. We might use a `<Cat mouse={{ x, y }}>` prop to tell the component the coordinates of the mouse so it knows where to position the image on the screen.
+
+As a first pass, you might try rendering the `<Cat>` *inside `<Mouse>`'s `render` method*, like this:
+
+```js
+class Cat extends React.Component {
+  render() {
+    const mouse = this.props.mouse
+    return (
+      <img src="/cat.jpg" style={{ position: 'absolute', left: mouse.x, top: mouse.y }}/>
+    );
+  }
+}
+
+class MouseWithCat extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleMouseMove = this.handleMouseMove.bind(this);
+    this.state = { x: 0, y: 0 };
+  }
+
+  handleMouseMove(event) {
+    this.setState({
+      x: event.clientX,
+      y: event.clientY
+    });
+  }
+
+  render() {
+    return (
+      <div style={{ height: '100%' }} onMouseMove={this.handleMouseMove}>
+
+        {/*
+          We could just swap out the <p> for a <Cat> here ... but then
+          we would need to create a separate <MouseWithSomethingElse>
+          component every time we need to use it, so <MouseWithCat>
+          isn't really reusable yet.
+        */}
+        <Cat mouse={this.state}/>
+      </div>
+    );
+  }
+}
+
+class MouseTracker extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Move the mouse around!</h1>
+        <MouseWithCat/>
+      </div>
+    );
+  }
+}
+```
+
+This approach will work for our specific use case, but we haven't achieved the objective of truly encapsulating the behavior in a reusable way. Now, every time we want the mouse position for a different use case, we have to create a new component (i.e. essentially another `<MouseWithCat>`) that renders something specifically for that use case.
+
+Here's where the render prop comes in: Instead of hard-coding a `<Cat>` inside a `<Mouse>` component, and effectively changing its rendered output, we can provide `<Mouse>` with a function prop that it uses to dynamically determine what to render–a render prop.
+
+```js
+class Cat extends React.Component {
+  render() {
+    const mouse = this.props.mouse
```

Comment:
you're missing a `;`

or better yet, destructure:
```js
const { mouse }  = this.props;
``

Reply:
Thanks, I'll gladly add the missing semi. A quick glance at the rest of the docs doesn't reveal any other usage of destructuring, so I'd rather not put it in this doc.

Reply:
@donavon - Since it's only used once, there's not really any advantage of destructuring here.

Reply:
@Daniel15, 

1. It's used twice. `left: mouse.x` and `top: mouse.y`.
1. Tighter code. This code is more concise:
    ```js
    const { mouse }  = this.props;
    ```
    that this code below. See how the term `mouse` is entered twice below?
    ```
    const mouse = this.props.mouse;
    ```
1. Consistency. In the next component that you write, you may pass props `foo` and `bar`. That should look the same.
    ```js
    const { foo, bar }  = this.props;
    ```

Reply:
I'm just trying to be more consistent with the rest of the docs. I don't see any others that use destructuring.